photon-ml/src/main/scala/com/linkedin/photon/ml/evaluation/PoissonLossEvaluator.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.evaluation
+
+import org.apache.spark.rdd.RDD
+
+import com.linkedin.photon.ml.function.PointwisePoissonLossFunction
+
+/**
+ * Evaluator for Poisson loss
+ *
+ * @param labelAndOffsetAndWeights a [[RDD]] of (id, (labels, offsets, weights)) pairs
+ * @param defaultScore the default score used to compute the metric
+ */
+protected[ml] class PoissonLossEvaluator(
+    labelAndOffsetAndWeights: RDD[(Long, (Double, Double, Double))],
+    defaultScore: Double = 0.0) extends Evaluator {
+
+  /**
+   * Evaluate the scores of the model
+   *
+   * @param scores the scores to evaluate
+   * @return score metric value
+   */
+  override def evaluate(scores: RDD[(Long, Double)]): Double = {
+    val defaultScore = this.defaultScore
+
+    val scoreAndLabelAndWeights = scores
+      .rightOuterJoin(labelAndOffsetAndWeights)
+      .mapValues { case (scoreOption, (label, offset, weight)) =>
+        (scoreOption.getOrElse(defaultScore) + offset, (label, weight))
+      }
+      .values
+
+    scoreAndLabelAndWeights.map { case (score, (label, weight)) =>
+        weight * PointwisePoissonLossFunction.loss(score, label)._1
```

Comment:
Are you using the same unit test for PoissonGradient evaluation that Brendan was using? Also, it is mandatory that the users only run Poisson regression with normalized features. Should this be specified as a comment in this file as well?